### PR TITLE
add optimization overview to sidebar and refresh

### PR DESCRIPTION
**Pull request by rosieyohannan**
# Description

I thought this would make a good reference page. The original ask in the ticket was to split this page up and make overview sections for the three optimization sections we have. I felt this wasn't the right thing to do at this time as we've been talking about making landing pages instead, so thought a reference page would work instead to find a place for the page in the sidebar.

# Reasons
Optimizations overview was not on the sidebar.

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

**Review by kbatuigas — CHANGES_REQUESTED** [review]
Couple of edits

File: jekyll/_cci2/optimizations.md [hunk 1]
```diff
@@ -1,36 +1,47 @@
...
+{: #speed}
+
+### Docker image choice
 {: #docker-image-choice }
 
 Choosing the right docker image for your project can have huge impact on build time. For example, choosing a basic language image means dependencies and tools need to be downloaded each time your pipeline is run, whereas, if you choose or build an image that has these dependencies and tools already installed, this time will be saved for each build run. When configuring your projects and specifying images, consider the following options:
 
-* CircleCI provides a range of [convenience images]({{site.baseurl}}/circleci-images/#section=configuration), typically based on official Docker images, but with a range of useful language tools pre-installed.
-* You can create [custom images]({{site.baseurl}}/custom-images/#section=configuration), maximizing specificity for your projects. To help with this we provide [guidance for building images manually]({{site.baseurl}}/custom-images/#creating-a-custom-image-manually).
+* CircleCI provides a range of [convenience images](/docs/circleci-images/#section=configuration), typically based on official Docker images, but with a range of useful language tools pre-installed.
```
Comment (rosieyohannan) [inline jekyll/_cci2/optimizations.md:41]:
```suggestion
* CircleCI provides a range of [convenience images](/docs/circleci-images/), typically based on official Docker images, but with a range of useful language tools pre-installed.
```

File: jekyll/_cci2/optimizations.md [hunk 2]
```diff
@@ -1,36 +1,47 @@
...
+* You can create [custom images](/docs/custom-images/#section=configuration), maximizing specificity for your projects. To help with this we provide [guidance for building images manually](/docs/custom-images/#creating-a-custom-image-manually).
```
Comment (rosieyohannan) [inline jekyll/_cci2/optimizations.md:42]:
```suggestion
* You can create [custom images](/docs/custom-images/), maximizing specificity for your projects. To help with this we provide [guidance for building images manually](/docs/custom-images/#creating-a-custom-image-manually).
```

File: jekyll/_cci2/optimizations.md [hunk 3]
```diff
@@ -39,46 +50,64 @@ DLC is similar to _caching dependencies_, in that it _saves_ the image layers th
...
 
 Using `resource_class`, it is possible to specify CPU and RAM resources for each job. For a full list of available resource class options for CircleCI cloud see the [configuration reference](/docs/configuration-reference/#resourceclass). For an equivalent list for CircleCI server installations, contact your system administrator.
 
 * Read more about resource classes on the [resource class overview](/docs/resource-class-overview/) page.
 
+## Optimize your configuration files
+{: #configuraiton }
+
+### Dynamic configuration
+{: dynamic-configuration}
```
Comment (rosieyohannan) [inline jekyll/_cci2/optimizations.md:85]:
```suggestion
{: #dynamic-configuration}
```

**Review by kbatuigas — APPROVED** [review]
🙌

**Pull request merged**